src/main.cpp

```diff
@@ -13,55 +13,219 @@
+                }
+            }
+        }
+        else {
+            printf("Failed to start rmdlconv process. Error code: %d\n", GetLastError());
+        }
+
+        // Clean up rmdlconv_out directory
+        fs::remove_all(rmdlconv_out_dir);
+    }
 }
 
 
 int main(int argc, char* argv[]) {
-    if (argc != 3) {
+    if (argc != 4) {
         print_usage(argv[0]);
         return 0;
     }
-    char* in_filename  = argv[1];
-    char* out_filename = argv[2];
+    char* game_dir = argv[1];
+    char* in_filename = argv[2];
+    char* output_folder_with_extension = argv[3];
+
+    // Remove .bsp extension from in_filename if present
+    std::string in_filename_without_ext(in_filename);
+    if (in_filename_without_ext.length() > 4 && in_filename_without_ext.substr(in_filename_without_ext.length() - 4) == ".bsp") {
+        in_filename_without_ext = in_filename_without_ext.substr(0, in_filename_without_ext.length() - 4);
+    }
+    // Ditto
+    std::string output_folder(output_folder_with_extension);
+    if (output_folder.length() > 4 && output_folder.substr(output_folder.length() - 4) == ".bsp") {
+        output_folder = output_folder.substr(0, output_folder.length() - 4);
+    }
+    // Initialize the file system with the provided game directory
+    if (!InitFileSystem(game_dir, in_filename)) {
+        fprintf(stderr, "Failed to initialize filesystem\n");
+        return 1;
+    }
+
+    // Create output directories
+    fs::path maps_output_dir = fs::path(output_folder) / "maps";
+    try {
+        fs::create_directories(maps_output_dir);
+    }
+    catch (const fs::filesystem_error& e) {
+        std::cerr << "Failed to create output directory: " << e.what() << std::endl;
+        return 1;
+    }
+
+    // Construct output filename
+    fs::path out_filename = maps_output_dir / fs::path(in_filename).filename();
+
+    std::cout << "Output file will be written to: " << out_filename << std::endl;
 
     int ret = 0;
+    std::vector<std::string> model_paths;
     try {
-        int convert(char* in_filename, char* out_filename);
-        ret = convert(in_filename, out_filename);
-    } catch (std::exception &e) {
-        fprintf(stderr, "Exception: %s\n", e.what());
+        int convert(char* in_filename, const char* out_filename, std::vector<std::string>&model_paths);
+        ret = convert(in_filename, fs::absolute(out_filename).string().c_str(), model_paths);
+    }
+    catch (const std::exception& e) {
+        fprintf(stderr, "Exception during conversion: %s\n", e.what());
         return 1;
     }
+
+    if (ret == 0) {
+        // Copy model files
+        copy_model_files(game_dir, output_folder.c_str(), model_paths, in_filename_without_ext);
+    }
+    else {
+        fprintf(stderr, "Conversion failed with return code: %d\n", ret);
+    }
+
     return ret;
 }
 
-
 void addPropsToCmGrid(
-    Bsp                              &r1bsp,
```

Comment:
please leave the variable initialisations alone so it's clear what's changed
same with number of lines between functions
helps with readability

Reply:
we're using `type &var;` because `type& var1, var2;` initialises `type &var1` and `type var2`
the only time I don't use this form is for `char*`, since I consider that to be it's own type (`string`)

the alignment with the spaces is to make it read more like a table
makes it easier to scan visually